plugins/volumeFilter_test.go

```diff
@@ -145,3 +146,276 @@ func TestMakeFilterVolume(t *testing.T) {
 		}
 	}
 }
+
+func TestVolumeFilterFn(t *testing.T) {
+	testCases := []struct {
+		name               string
+		mode               volumeFilterMode
+		sellBaseCapInBase  *float64
+		sellBaseCapInQuote *float64
+		otbBaseCap         *float64
+		otbQuoteCap        *float64
+		tbbBaseCap         *float64
+		tbbQuoteCap        *float64
+		price              string
+		inputAmount        string
+		wantAmount         string
+		wantTbbBaseCap     *float64
+		wantTbbQuoteCap    *float64
+	}{
+		{
+			name:               "selling, base units sell cap, don't keep selling base, exact mode",
+			mode:               volumeFilterModeExact,
+			sellBaseCapInBase:  pointy.Float64(0.0),
+			sellBaseCapInQuote: nil,
+			otbBaseCap:         pointy.Float64(0.0),
+			otbQuoteCap:        pointy.Float64(0.0),
+			tbbBaseCap:         pointy.Float64(0.0),
+			tbbQuoteCap:        pointy.Float64(0.0),
+			price:              "2.0",
+			inputAmount:        "100.0",
+			wantAmount:         "",
```

Comment:
Got it - this is what I was originally thinking, but I changed the approach because of [this previous comment](https://github.com/stellar/kelp/pull/552#discussion_r512331838). Which approach should I follow?

Reply:
we should follow the approach suggested in this PR, thanks for checking 👍